modules/phase_field/src/interfacekernels/InterfaceGradientMatch.C

```diff
@@ -0,0 +1,61 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "InterfaceGradientMatch.h"
+
+template<>
+InputParameters validParams<InterfaceGradientMatch>()
+{
+  InputParameters params = validParams<InterfaceKernel>();
+  params.addClassDescription("Enforce gradient continuity between two different variables across a subdomain boundary");
+  params.addRequiredParam<unsigned int>("component", "Gradient component to set equal");
+  return params;
+}
+
+InterfaceGradientMatch::InterfaceGradientMatch(const InputParameters & parameters) :
+    InterfaceKernel(parameters),
+    _component(getParam<unsigned int>("component"))
+{
+}
+
+Real
+InterfaceGradientMatch::computeQpResidual(Moose::DGResidualType type)
+{
+  // equal gradients means difference is zero
+  Real res = _grad_u[_qp](_component) - _grad_neighbor_value[_qp](_component);
+
```

Comment:
More general way is to subtract along the normal direction of the surface, I guess. Do you assume that the boundary will be aligned  with one of axis directions?